pennylane/transforms/metric_tensor.py

```diff
@@ -14,105 +14,140 @@
 """
 Contains the metric tensor transform
 """
+import functools
 import warnings
 
 import numpy as np
 import pennylane as qml
 
 
+from .batch_transform import batch_transform
+
+
+SUPPORTED_OPS = ["RX", "RY", "RZ", "PhaseShift"]

```

Comment:
For the metric tensor, we can support any single-parameter operation that has a generator, so this limits us to these four (I believe?)

As to why `PhaseShift`; on the off-chance someone _does_ have a circuit with `PhaseShift`, I think this will lead to a slight reduction in overhead, since the expansion can be avoided.